motor.cpp

```diff
@@ -23,7 +23,22 @@ void Motor::stop(int motor, int motor2, int motor3, int motor4) {
     }
 }
 
+void Motor::go_all(
+    int motor1_speed, int motor2_speed,
+    int motor3_speed, int motor4_speed,
+    int motor1_dir, int motor2_dir,
+    int motor3_dir, int motor4_dir) {
+
+    Motor::go(1, motor1_dir, motor1_speed = (motor1_speed < 0)?0:motor1_speed);
```

Comment:
This is not Python, so the third argument would be only `motor1_speed < 0 ? 0 : motor1_speed`, also this is unnecesary, since you added that check to `go` as well as `go_pair` and `rotate` functions.